nifi-stardog-processors/src/main/java/com/stardog/nifi/StardogPut.java

```diff
@@ -386,6 +410,11 @@ else if (filename.toLowerCase().endsWith("json")) {
                 vgConn.importFile(mappingString, properties, connection.name(), targetGraph, in, fileType);
             }
             else {
+                if (!clearTargetGraph && queryStr != null) {
+                    UpdateQuery updateQuery = connection.update(queryStr);
+                    updateQuery.execute();
+                    logger.info("Update query successfully executed");
+                }
```

Comment:
Same thoughts here about exclusivity, though in this case the clear happens within the transaction below. So in this branch it does seem the two options are incompatible and so it may be better to add a validation rule.